src/createServer.js

```diff
@@ -1,3 +1,74 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+/* eslint-disable max-len */
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+const { validCases } = require('./vars/cases');
+const {
+  noTextError,
+  noToCaseError,
+  invalidCaseError,
+} = require('./vars/errors');
+let result;
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    const { pathname, searchParams } = new URL(
+      'http://localhost:3000' + req.url,
+    );
+    const textToConvert = pathname.substring(1);
+    const toCase = searchParams.get('toCase');
+    const errors = [];
+
+    if (!textToConvert) {
+      errors.push({
+        message: noTextError,
+      });
+    }
+
+    if (!toCase) {
+      errors.push({
+        message: noToCaseError,
+      });
+    }
+
+    if (toCase && !validCases.includes(toCase.toUpperCase())) {
+      errors.push({
+        message: invalidCaseError,
+      });
+    }
+
+    if (errors.length > 0) {
+      res.writeHead(400, {
+        'Content-Type': 'application/json',
+      });
+
+      res.end(JSON.stringify({ errors }));
+
+      return;
+    }
+
+    try {
+      result = convertToCase(textToConvert, toCase);
+
+      res.writeHead(200, {
+        'Content-Type': 'application/json',
+      });
+    } catch (err) {
+      throw err;
```

Comment:
The `catch` block rethrows the error without handling it. This could lead to server crashes if an unexpected error occurs. Consider sending a 500 status code with an error message to the client instead.